frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -828,27 +835,55 @@ function ReaderFooter:addToMainMenu(menu_items)
+                    {
+                        text_func = function()
+                            return T(_("Font size (%1)"), self.settings.text_font_size)
+                        end,
+                        callback = function(touchmenu_instance)
+                            local SpinWidget = require("ui/widget/spinwidget")
+                            local font_size = self.settings.text_font_size
+                            local items_font = SpinWidget:new{
+                                width = Screen:getWidth() * 0.6,
+                                value = font_size,
+                                value_min = 8,
+                                value_max = 36,
+                                default_value = 14,
+                                ok_text = _("Set size"),
+                                title_text =  _("Footer font size"),
+                                callback = function(spin)
+                                    self.settings.text_font_size = spin.value
+                                    --local text = self.footer_text.text
+                                    self.footer_text = TextWidget:new{
+                                        text = self.footer_text.text,
+                                        face = Font:getFace(self.text_font_face, self.settings.text_font_size),
+                                        bold = self.settings.text_font_bold,
+                                    }
+                                    self.text_container = RightContainer:new{
+                                        dimen = Geom:new{ w = 0, h = self.height },
+                                        self.footer_text,
+                                    }
+                                    self:refreshFooter(true, true)
+                                    if touchmenu_instance then touchmenu_instance:updateItems() end
+                                end,
+                            }
+                            UIManager:show(items_font)
+                        end,
+                        keep_menu_open = true,
+                    },
+                    {
+                        text = _("Bold"),
+                        checked_func = function()
+                            return self.settings.text_font_bold == true
+                        end,
+                        callback = function(touchmenu_instance)
+                            self.settings.text_font_bold = not self.settings.text_font_bold
                             self.footer_text = TextWidget:new{
-                                text = text,
-                                face = Font:getFace(self.text_font_face, self.settings.text_font_size)
+                                text = self.footer_text.text,
+                                face = Font:getFace(self.text_font_face, self.settings.text_font_size),
+                                bold = self.settings.text_font_bold,
                             }
```

Comment:
Here and above:
- please call `self.footer_text:free()` before replacing it, to release some resources (xtext) used by the old one - it would be free'd by the Garbage Collector, but it's best to think about doing it explicitely. (When the widget is closed by UIManager as all usually are, no need to do it explicitely.)
- a bit unfortunate to have that RightContainer re-created here (if we change some stuff at the original place, we'll probably forget to change it here) - I think just doing: `self.text_container[1] = self.footer_text` should work
- I would say "a bit unfortunate to have that TextWidget re-created here" but well, it's quite simple, so that's ok (the alternative would have been to add a `TextWidget:setFace(face, bold)` like the existing setText/setMaxWidth, but you'd probably be the only one bit of code that would use it, so, well, let's keep it that way :)

Reply:
Thanks. Done :)